src/test/java/com/synopsys/integration/detect/lifecycle/run/step/ContainerScanStepRunnerTest.java

```diff
@@ -0,0 +1,113 @@
+import com.synopsys.integration.detect.workflow.file.DirectoryManager;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.util.NameVersion;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ContainerScanStepRunnerTest {
+    @Mock
+    private ContainerUploader mockContainerUploader;
+    
+    @Mock
+    private File mockContainerImage;
+    
+    private ContainerScanStepRunner instance;
+    
+    @BeforeAll
+    public void initMocks() throws Exception {
+        MockitoAnnotations.openMocks(this);
+        
+        NameVersion mockNameVersion = mock(NameVersion.class);
+        OperationRunner mockOperationRunner = mock(OperationRunner.class);
+        DirectoryManager mockDirectoryManager = mock(DirectoryManager.class);
+        File mockBinaryOutputDirectory = mock(File.class);
+        ProjectGroupOptions mockProjectGroupOptions = mock(ProjectGroupOptions.class);
+        BlackDuckRunData mockBlackDuckRunData = mock(BlackDuckRunData.class);
+        BlackDuckVersion mockBlackDuckVersion = mock(BlackDuckVersion.class);
+        UploaderFactory mockUploadFactory = mock(UploaderFactory.class);
+
+        when(mockOperationRunner.calculateProjectGroupOptions()).thenReturn(mockProjectGroupOptions);
+        String expectedProjectGroupName = "expectedProjectGroupName";
+        when(mockProjectGroupOptions.getProjectGroup()).thenReturn(expectedProjectGroupName);
+        when(mockOperationRunner.getDirectoryManager()).thenReturn(mockDirectoryManager);
+        when(mockDirectoryManager.getBinaryOutputDirectory()).thenReturn(mockBinaryOutputDirectory);
+        when(mockBinaryOutputDirectory.exists()).thenReturn(true);
+        when(mockBlackDuckRunData.getBlackDuckServerVersion()).thenReturn(Optional.of(mockBlackDuckVersion));
+        
+        // We are going to say false here to avoid mocking and testing a whole bunch of library specific init
+        // code that the ContainerScanStepRunner constructor does.
+        // We'll call the method we are after directly in the method.invoke call below.
+        when(mockBlackDuckVersion.isAtLeast(new BlackDuckVersion(2024, 10, 0))).thenReturn(false);
+        
+        when(mockUploadFactory.createContainerUploader(Mockito.anyString())).thenReturn(mockContainerUploader);
+        
+        when(mockOperationRunner.getContainerScanImage(Mockito.any(), Mockito.any())).thenReturn(mockContainerImage);
+     
+        instance = new ContainerScanStepRunner(mockOperationRunner, mockNameVersion, mockBlackDuckRunData, new Gson());
+        
+        Field fieldScan = ContainerScanStepRunner.class.getDeclaredField("scanId");
+        fieldScan.setAccessible(true);
+        UUID scanId = UUID.randomUUID();
+        fieldScan.set(instance, scanId);
+
+        Field fieldFactory = ContainerScanStepRunner.class.getDeclaredField("uploadFactory");
+        fieldFactory.setAccessible(true);
+        fieldFactory.set(instance, mockUploadFactory);
```

Comment:
My first time seeing private methods/variables be accessed like this, got to learn something new reviewing this!

Reply:
The test definitely got into the realm of am I mocking too much...most of this is a library call so the heavy testing probably should be in there, still, hopefully it provides some basic value.